cocos/particle/renderer/particle-system-renderer-data.ts

```diff
@@ -203,11 +203,24 @@ export default class ParticleSystemRenderer {
 
     private _particleSystem: any = null!; // ParticleSystem
 
+    create (ps: any) {
+        if (this._particleSystem === null) {
+            this._particleSystem = ps;
+            const useGPU = this._useGPU && isSupportGPUParticle();
+            this._particleSystem.processor = useGPU ? new ParticleSystemRendererGPU(this) : new ParticleSystemRendererCPU(this);
+            this._particleSystem.processor.onInit(this._particleSystem);
+        }
+    }
+
     onInit (ps: any) {
-        this._particleSystem = ps;
-        const useGPU = this._useGPU && isSupportGPUParticle();
-        this._particleSystem.processor = useGPU ? new ParticleSystemRendererGPU(this) : new ParticleSystemRendererCPU(this);
-        this._particleSystem.processor.onInit(ps);
+        // if particle system is null we run the old routine
+        // else if particle system is not null we do nothing
+        if (this._particleSystem === null) {
+            this._particleSystem = ps;
+            const useGPU = this._useGPU && isSupportGPUParticle();
+            this._particleSystem.processor = useGPU ? new ParticleSystemRendererGPU(this) : new ParticleSystemRendererCPU(this);
+            this._particleSystem.processor.onInit(ps);
+        }
```

Comment:
```
onInit (ps: any) {
    // No repetitive code
    this.create(ps);
    if (!this._particleSystem.processor) {
        const useGPU = this._useGPU && isSupportGPUParticle();
        this._particleSystem.processor = useGPU ? new ParticleSystemRendererGPU(this) : new ParticleSystemRendererCPU(this);
        this._particleSystem.processor.onInit(ps);
    }
    else {
        // message: Particle system shouldn't be initialized repetitively, otherwise there will be potential leak.
        errorID(...);
    }
}
```